### DISCUSSION TITLE
QSMART

### DISCUSSION THREAD

**Discussion by mgreen-neura** [neurodesk/neurodesk.github.io-hugo#283] [Ideas]
Hi all,

I'm just wondering if it's possible to include this QSMART analysis software on NeuroDesk at all?
No worries if not or if it's not possible but it'd be incredibly helpful to add it to the suite of NeuroDesk software.

https://github.com/MBCIU/QSMART

Kind Regards,
Michael

**Comment by stebo85** [comment]
Dear @mgreen-neura - that's possible. The only problem is that it needs Matlab, so users would need a Matlab license. You could start from this recipe https://github.com/NeuroDesk/neurocontainers/blob/master/recipes/matlab/build.sh - let me know if you need any help :)

**Comment by civier** [comment]
Hi @mgreen-neura 

Another option is to compile QSMART using the Matlab compiler, and then include the compiled code inside a Neurodesk container. This is what I did with the code of my recent MFCSC method, which is now in Neurodesk as well:
https://github.com/NeuroDesk/neurocontainers/tree/master/recipes/mfcsc
The advantage of compiling the code is that users won't need Matlab license at all. The disadvantages are that users will be able to interact with the compiled QSMART only via command-line, GUI (if it has one), or by writing a single-file Matlab script that calls to other functions in QSMART. If it would work for QSMART, I'd be happy to assist you in compiling it. It's quite tricky, but luckily I've just done it recently.

Best,
Oren

**Reply by stebo85** [reply]
This could work, the only problem is that QSMART uses pfiles that obfuscate the matlab code. It may not work to compile it?

**Comment by stebo85** [comment]
Dear @mgreen-neura,

I just tried out to run QSMART on Neurodesk and it already works quite well. You just need to download the matlab toolboxes as described in the Readme. FSL and ANTS are available through the module system.

It might not need a new container? There is another problem with STI SUITE: It's not quite open source because you need to register to get access to the code. This might complicate redistributing it.

**Comment by mgreen-neura** [comment]
Thanks for the responses. I'll have a go at running it on Neurodesk as you did and see how it goes! Thanks for the help.
